web-client/public/js/upload.js

```diff
@@ -138,7 +139,16 @@ export const upload = function () {
                         stopLoadingIcon();
                         if (Object.keys(exportSheets).length === chosenSheets.length) {
                             // we have all of the sheets so lets initilize the export process
-                            grnState.workbook.exportExpression = exportSheets;
+                            const finalExportSheets = {};
+                            Object.keys(exportSheets).forEach((sheet) => {
+                                // make sure that the sheets we queried are populated with the correct data
+                                if (exportSheets[sheet].data && exportSheets[sheet].timePoints) {
+                                    // if the resulting query contains both the timePoint data and
+                                    // the gene data then export it. If not don't :)
```

Comment:
Documentation and clarity is always good! I love how you specified in this section, as it is a large block of code.